client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/ContactsSection/GroupedInvestigations/GroupedInvestigationForm/ContactsForm/InvestigationAccordion/AccordionContent/ContactsTable/ContactsTable.tsx

```diff
@@ -25,6 +26,7 @@ interface Props {
     selectedRows : number[];
     setSelectedRows: React.Dispatch<React.SetStateAction<number[]>>;
     events : ContactEvent[];
+    existingIds: string[];
```

Comment:
Maybe use context instead of paying through levels who don't use it?

Reply:
Maybe slecetd rows as well

Reply:
Could refer to family involvement context, which does something similar 🤔🤔

Reply:
Introducing Context is very wasteful, it re-renders the entire tree (so in that case the entire form) instead of re-rendering only the appropriate Components whenever the state changes.
I figured because the selectedRows needs to eventually be addressed at the top level (at the sending of the form)
and the existingIds exist on the top level and needs to get to the lowest level that those fields are being used in a very narrow way so moving them into a provider will cause a lot of re-rendering (from all of the 'branches' of the tree)

Reply:
Plus the family member context only refers to the current users family members , what Im requiring here is all of the event's IDs and if the relation is family when Im importing another users family member

Reply:
my understanding was different, so i would love if you could elaborate cause i might be wrong

from [reacts site](https://reactjs.org/docs/context.html#consumer):
> All **consumers that are descendants of a Provider** will re-render whenever the Provider’s value prop changes. The propagation from Provider to its descendant consumers (including .contextType and useContext) is not subject to the shouldComponentUpdate method, so the consumer is updated even when an ancestor component skips an update.

Reply:
also,
* i would probably change the prop's name 🤷‍♀️ it took me a while to figure out what `events` was...
* if allInvestigationIds is only used to create exisitingIds maybe just parse it to be exisitingIds and pass it that way? or am i missing something? 🤔

Reply:
renamed the props name to Investigation [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1246/commits/26a9218340a706c5cf2be132e035fe93570f18be)

allContactsIds is being used to check for duplicate IDs in the onSubmit functions and contain all of the users contacts

Reply:
Moved to context [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1246/commits/14d1ec18b3cd76bb8f50e95a46a48613adf4967e) and [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1246/commits/21abd99e7e8c3c21feb162e83d1c06366ae98538)